linuxgsm.sh

```diff
@@ -401,50 +401,63 @@ else
-			source "${configdirserver}/${selfname}.cfg"
-		else
-			# shellcheck source=/dev/null
-			source "${configdirserver}/${selfname}.cfg"
-		fi
-		# Load the secrets-instance.cfg config. If missing download it.
-		if [ ! -f "${configdirserver}/secrets-${selfname}.cfg" ]; then
-			fn_fetch_config "lgsm/config-default/config-lgsm" "secrets-instance-template.cfg" "${configdirserver}" "secrets-${selfname}.cfg" "nochmodx" "norun" "noforcedl" "nomd5"
-			# shellcheck source=/dev/null
-			source "${configdirserver}/secrets-${selfname}.cfg"
-		else
-			# shellcheck source=/dev/null
-			source "${configdirserver}/secrets-${selfname}.cfg"
-		fi
+			source "${configdirserver}/_default.cfg"
+			# Load the common.cfg config. If missing download it.
+			if [ ! -f "${configdirserver}/common.cfg" ]; then
+				fn_fetch_config "lgsm/config-default/config-lgsm" "common-template.cfg" "${configdirserver}" "common.cfg" "${chmodx}" "nochmodx" "norun" "noforcedl" "nomd5"
+				# shellcheck source=/dev/null
+				source "${configdirserver}/common.cfg"
+			else
+				# shellcheck source=/dev/null
+				source "${configdirserver}/common.cfg"
+			fi
+			# Load the secrets-common.cfg config. If missing download it.
+			if [ ! -f "${configdirserver}/secrets-common.cfg" ]; then
+				fn_fetch_config "lgsm/config-default/config-lgsm" "secrets-common-template.cfg" "${configdirserver}" "secrets-common.cfg" "${chmodx}" "nochmodx" "norun" "noforcedl" "nomd5"
+				# shellcheck source=/dev/null
+				source "${configdirserver}/secrets-common.cfg"
+			else
+				# shellcheck source=/dev/null
+				source "${configdirserver}/secrets-common.cfg"
+			fi
+			# Load the instance.cfg config. If missing download it.
+			if [ ! -f "${configdirserver}/${selfname}.cfg" ]; then
+				fn_fetch_config "lgsm/config-default/config-lgsm" "instance-template.cfg" "${configdirserver}" "${selfname}.cfg" "nochmodx" "norun" "noforcedl" "nomd5"
+				# shellcheck source=/dev/null
+				source "${configdirserver}/${selfname}.cfg"
+			else
+				# shellcheck source=/dev/null
+				source "${configdirserver}/${selfname}.cfg"
+			fi
+			# Load the secrets-instance.cfg config. If missing download it.
+			if [ ! -f "${configdirserver}/secrets-${selfname}.cfg" ]; then
+				fn_fetch_config "lgsm/config-default/config-lgsm" "secrets-instance-template.cfg" "${configdirserver}" "secrets-${selfname}.cfg" "nochmodx" "norun" "noforcedl" "nomd5"
+				# shellcheck source=/dev/null
+				source "${configdirserver}/secrets-${selfname}.cfg"
+			else
+				# shellcheck source=/dev/null
+				source "${configdirserver}/secrets-${selfname}.cfg"
+			fi
+			if [ -n "$(grep startparameters "${configdirserver}/common.cfg" | sed -e '/^#/d')" ]||[ -n "$(grep startparameters "${configdirserver}/${selfname}.cfg" | sed -e '/^#/d')" ]||[ -n "$(grep startparameters "${configdirserver}/secrets-common.cfg" | sed -e '/^#/d')" ]||[ -n "$(grep startparameters "${configdirserver}/secrets-${selfname}.cfg" | sed -e '/^#/d')" ]; then
```

Comment:
I guess it would be also possible to check all files with one grep, as this only checks if the the startparameters are in the configfile, but in the end this depends on the coding style.

Reply:
I would for this only check the common and instance config.

Reply:
Happy to keep as it is. There might be some reason they want to have startparameters in the secret config